packages/selected-network-controller/src/SelectedNetworkMiddleware.ts

```diff
@@ -1,65 +1,36 @@
-import type { ControllerMessenger } from '@metamask/base-controller';
 import type { JsonRpcMiddleware } from '@metamask/json-rpc-engine';
-import type {
-  NetworkClientId,
-  NetworkControllerGetStateAction,
-  NetworkControllerStateChangeEvent,
-} from '@metamask/network-controller';
+import type { NetworkClientId } from '@metamask/network-controller';
 import type { Json, JsonRpcParams, JsonRpcRequest } from '@metamask/utils';
 
-import type {
-  SelectedNetworkControllerGetNetworkClientIdForDomainAction,
-  SelectedNetworkControllerSetNetworkClientIdForDomainAction,
-} from './SelectedNetworkController';
+import type { SelectedNetworkControllerMessenger } from './SelectedNetworkController';
 import { SelectedNetworkControllerActionTypes } from './SelectedNetworkController';
 
-export type MiddlewareAllowedActions = NetworkControllerGetStateAction;
-export type MiddlewareAllowedEvents = NetworkControllerStateChangeEvent;
-
-export type SelectedNetworkMiddlewareMessenger = ControllerMessenger<
-  | SelectedNetworkControllerGetNetworkClientIdForDomainAction
-  | SelectedNetworkControllerSetNetworkClientIdForDomainAction
-  | MiddlewareAllowedActions,
-  MiddlewareAllowedEvents
->;
-
 export type SelectedNetworkMiddlewareJsonRpcRequest = JsonRpcRequest & {
   networkClientId?: NetworkClientId;
   origin?: string;
 };
 
 export const createSelectedNetworkMiddleware = (
-  messenger: SelectedNetworkMiddlewareMessenger,
+  messenger: SelectedNetworkControllerMessenger,
 ): JsonRpcMiddleware<JsonRpcParams, Json> => {
   const getNetworkClientIdForDomain = (origin: string) =>
     messenger.call(
       SelectedNetworkControllerActionTypes.getNetworkClientIdForDomain,
       origin,
     );
-
-  const setNetworkClientIdForDomain = (
-    origin: string,
-    networkClientId: NetworkClientId,
-  ) =>
+  const getNetworkClientIdForMetamask = () =>
     messenger.call(
-      SelectedNetworkControllerActionTypes.setNetworkClientIdForDomain,
-      origin,
-      networkClientId,
+      SelectedNetworkControllerActionTypes.getNetworkClientIdForMetamask,
     );
 
-  const getDefaultNetworkClientId = () =>
-    messenger.call('NetworkController:getState').selectedNetworkClientId;
-
   return (req: SelectedNetworkMiddlewareJsonRpcRequest, _, next) => {
     if (!req.origin) {
       throw new Error("Request object is lacking an 'origin'");
     }
 
-    if (getNetworkClientIdForDomain(req.origin) === undefined) {
-      setNetworkClientIdForDomain(req.origin, getDefaultNetworkClientId());
-    }
-
-    req.networkClientId = getNetworkClientIdForDomain(req.origin);
+    req.networkClientId =
+      getNetworkClientIdForDomain(req.origin) ??
+      getNetworkClientIdForMetamask();
```

Comment:
We no longer want to add the requesting origin to the `SelectedNetworkController` domains state if not previously set - this led to an issue where any and all sites the user visited were added to this `domains` state. Rather we ensure the origin is added to state when the dapp is granted permissions and if its not connected we fallback to adding the globally selected networkClientId on the request.